ui/imports/shared/views/ProfileDialogView.qml

```diff
@@ -0,0 +1,663 @@
+                }
+            }
+
+            Loader {
+                Layout.alignment: Qt.AlignTop
+                Layout.preferredHeight: menuButton.visible ? menuButton.height : -1
+                sourceComponent: {
+                    // current user
+                    if (d.isCurrentUser)
+                        return btnEditProfileComponent
+
+                    // contact request, outgoing, rejected
+                    if (!d.isContact && d.isContactRequestSent && d.outgoingVerificationStatus === Constants.verificationStatus.declined)
+                        return txtRejectedContactRequestComponent
+                    // contact request, outgoing, pending
+                    if (!d.isContact && d.isContactRequestSent)
+                        return txtPendingContactRequestComponent
+
+                    // contact request, incoming, pending
+                    if (!d.isContact && d.isContactRequestReceived)
+                        return btnAcceptContactRequestComponent
+
+                    // contact request, incoming, rejected
+                    if (d.isContactRequestSent && d.incomingVerificationStatus === Constants.verificationStatus.declined)
+                        return btnBlockUserComponent
+
+                    // verified contact request, incoming, pending
+                    if (d.isContact && !d.isTrusted && d.isVerificationRequestReceived)
+                        return btnRespondToIdRequestComponent
+
+                    // block user
+                    if (!d.isContact && !d.isBlocked &&
+                            (d.contactDetails.trustStatus === Constants.trustStatus.untrustworthy || d.outgoingVerificationStatus === Constants.verificationStatus.declined))
+                        return btnBlockUserComponent
+
+                    // send contact request
+                    if (!d.isContact && !d.isBlocked && !d.isContactRequestSent)
+                        return btnSendContactRequestComponent
+
+                    // blocked contact
+                    if (d.isBlocked)
+                        return btnUnblockUserComponent
+
+                    // send message
+                    if (d.isContact && !d.isBlocked)
+                        return btnSendMessageComponent
+
+                    console.warn("!!! UNHANDLED CONTACT ACTION BUTTON; PUBKEY", root.publicKey)
+                }
+            }
+
+            StatusFlatButton {
+                id: menuButton
+                Layout.alignment: Qt.AlignTop
+                Layout.preferredWidth: height
+
+                visible: !d.isCurrentUser
+                size: StatusBaseButton.Size.Small
+                horizontalPadding: 6
+                verticalPadding: 6
+                icon.name: "more"
+                icon.color: Theme.palette.directColor1
+                highlighted: moreMenu.opened
+                onClicked: moreMenu.popup(-moreMenu.width + width, height + 4)
+
+                StatusPopupMenu {
```

Comment:
Just a suggestion: It could be written in a separate file.

Reply:
Yup, fully agree; right now it's a bit of a mess, it should be extracted and also used instead of `MessageContextMenuView` when clicking on a profile in the chat (and other parts of the app). Please file an issue :)

Reply:
Here the new issue to keep tracking: https://github.com/status-im/status-desktop/issues/7886